src/main/java/org/gridsuite/modification/server/modifications/VoltageInitModification.java

```diff
@@ -4,264 +4,304 @@
-            if (m.getVoltageSetpoint() != null || m.getReactivePowerSetpoint() != null) {
-                Reporter reporter = subReporter.createSubReporter(GENERATOR_MSG + m.getGeneratorId(), GENERATOR_MSG + m.getGeneratorId());
-                report(reporter, "generatorModification", "Generator with id=${id} modified :", Map.of("id", m.getGeneratorId()), TypedValue.INFO_SEVERITY);
+                Reporter reporter = subReporter.createSubReporter(GENERATOR_KEY, GENERATOR_NAME, "id", m.getGeneratorId());
+                reporter.report(Report.builder().withKey("generatorNotFound")
+                                                .withDefaultMessage("Generator with id=${id} not found")
+                                                .withValue("id", m.getGeneratorId())
+                                                .withSeverity(TypedValue.WARN_SEVERITY).build());
+            } else if (m.getVoltageSetpoint() != null || m.getReactivePowerSetpoint() != null) {
+                modificationsCount++;
+                Reporter reporter = subReporter.createSubReporter(GENERATOR_KEY, GENERATOR_NAME, "id", m.getGeneratorId());
+                reporter.report(Report.builder().withKey("generatorModification")
+                                                .withDefaultMessage("Generator with id=${id} modified :")
+                                                .withValue("id", m.getGeneratorId())
+                                                .withSeverity(TypedValue.TRACE_SEVERITY).build());
                 if (m.getVoltageSetpoint() != null) {
-                    double oldTargetV = generator.getTargetV();
+                    final double oldTargetV = generator.getTargetV();
                     generator.setTargetV(m.getVoltageSetpoint());
-                    reporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(oldTargetV, m.getVoltageSetpoint(), VOLTAGE_SET_POINT, 1));
+                    reporter.report(ModificationUtils.buildModificationReport(oldTargetV, m.getVoltageSetpoint(), VOLTAGE_SET_POINT, 1, TypedValue.TRACE_SEVERITY));
                 }
                 if (m.getReactivePowerSetpoint() != null) {
-                    double oldTargetQ = generator.getTargetQ();
+                    final double oldTargetQ = generator.getTargetQ();
                     generator.setTargetQ(m.getReactivePowerSetpoint());
-                    reporter.report(ModificationUtils.getInstance().buildModificationReportWithIndentation(oldTargetQ, m.getReactivePowerSetpoint(), REACTIVE_POWER_SET_POINT, 1));
+                    reporter.report(ModificationUtils.buildModificationReport(oldTargetQ, m.getReactivePowerSetpoint(), REACTIVE_POWER_SET_POINT, 1, TypedValue.TRACE_SEVERITY));
                 }
             }
-        });
+        }
+        if (modificationsCount > 0) {
+            subReporter.report(new Report("generatorModificationsResume", "${count} generator(s) have been modified.", Map.of(
```

Comment:
```suggestion
          String message = modificationsCount == 1 ? "${count} generator has been modified." : "${count} generators have been modified.";
            subReporter.report(new Report("generatorModificationsResume", message, Map.of(
```

Reply:
can't work as it because when reporter+report are serialized, a `defaultMessage` must always be the same for a `reportKey`.  
look at json logs in test resources